gprofiler/metadata/py_module_version.py

```diff
@@ -0,0 +1,266 @@
+    if metadata:
+        # The metadata should NEVER be missing the Name: key, but if it somehow
+        # does, fall back to the known canonical name.
+        return metadata.get("Name", dist.project_name)
+    return None
+
+
+def _get_libpython_path(pid: int) -> Optional[str]:
+    # Match the path to libpython as it may appear in /proc/[pid]/maps, e.g. " /usr/local/lib/libpython3.9.so.1.0"
+    libpython_maps_pattern = re.compile(r"(?<=\s)/\S*libpython\S*\.so(\.\S+)?\Z")
+
+    try:
+        f = open(f"/proc/{pid}/maps")
+    except OSError:
+        return None
+
+    with f:
+        for line in f.readlines():
+            match = libpython_maps_pattern.search(line.strip())
+            if match is not None:
+                return match.group()
+    return None
+
+
+@functools.lru_cache(maxsize=128)
+def _get_python_full_version(process: Process, short_version: str) -> Optional[str]:
+    assert re.match(r"[23]\.\d\d?", short_version)
+
+    bin_file = _get_libpython_path(process.pid) or f"/proc/{process.pid}/exe"
+    full_version_string_pattern = re.compile(rb"(?<=\D)" + short_version.encode() + rb"\.\d\d?(?=\x00)")
+
+    # Try to extract the version string from the binary
+    try:
+        f = open(resolve_host_path(process, bin_file), "rb")
+    except OSError:
+        return None
+
+    with f:
+        for line in f.readlines():
+            match = full_version_string_pattern.search(line)
+            if match is not None:
+                return match.group().decode()
+    return None
+
+
+def _get_standard_libs_version(result: Dict[str, Optional[Tuple[str, str]]], process: Process):
+    # Standard library modules are identified by being under a pythonx.y dir and *not* under site/dist-packages
+    standard_lib_pattern = re.compile(r"/python(?P<version>\d\.\d\d?)/(?!.*(site|dist)-packages)")
+    py_version = None
+
+    for path in result:
+        match = standard_lib_pattern.search(path)
+        if match is not None:
+            if py_version is None:
+                py_version = _get_python_full_version(process, match.group("version"))
+                if py_version is None:
+                    # No need to continue trying if we failed
+                    return None
+            result[path] = ("standard-library", py_version)  # type: ignore
+
+
+@functools.lru_cache(maxsize=128)
+def _get_dists_files(packages_path: str) -> Dict[str, pkg_resources.Distribution]:
+    """Return a dict of filename: dist for the distributions in packages_path"""
+    path_to_dist = {}
+    for dist in pkg_resources.find_distributions(packages_path):
+        files_iter = _files_from_record(dist) or _files_from_legacy(dist)
+        if files_iter is not None:
+            path_to_dist.update(dict.fromkeys((os.path.join(packages_path, file) for file in files_iter), dist))
+    return path_to_dist
+
+
+_warned_no__normalized_cached = False
+
+
+def _get_packages_versions(result: Dict[str, Optional[Tuple[str, str]]], process: Process):
```

Comment:
Any reason for getting the result as an argument and also returning it?
I don't see any reason getting it as an argument, you can just return it as a result and then merge all the results (the cost of this one is negligible

Reply:
The name `result` is very confusing in this context. 
1. It'd be better named `packages_versions`.
2. The function would be better named `populate_packages_versions` to signify that it modifies the parameter and does not return a value.